### ISSUE TITLE
Attempting to run spinning_cube on android causes exception.

### ISSUE THREAD

**Issue by smklein** [domokit/mojo#500]
To reproduce (on master):
./mojo/tools/mojob.py gn --android
./mojo/tools/mojob.py build --android
./mojo/devtools/common/mojo_run https://core.mojoapps.io/spinning_cube.mojo --android

With an Android device connected (which worked with spinning cube just days ago):
Exception in thread StdoutRedirector:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/google/home/smklein/ssd/mojo/src/mojo/devtools/common/devtoolslib/android_shell.py", line 136, in _run
    _wait_for_fifo()
  File "/usr/local/google/home/smklein/ssd/mojo/src/mojo/devtools/common/devtoolslib/android_shell.py", line 135, in _wait_for_fifo
    raise Exception("Unable to find fifo.")
Exception: Unable to find fifo.

**Comment by smklein** [comment]
Caused by:
https://codereview.chromium.org/1414633004

**Comment by smklein** [comment]
@qsr @ppiet

**Comment by nellyv** [comment]
It's a read/write permission problem we have for Marshmallow.

TL;DR: 
Go to Settings -> Apps -> Mojo Shell (dev) -> Permissions -> allow from storage. 
This is a temporary solution, we will fix it.

Long version: 
Starting in API level 23 (Marshmallow) the applications need to request read/write permissions at runtime. An exception to this rule is for files stored in the folder returned by method [Context.getExternalFilesDir(String)](http://developer.android.com/reference/android/content/Context.html#getExternalFilesDir(java.lang.String)) which can be used without permissions. This folder looks like "/storage/emulated/0/Android/data/org.chromium.mojo.shell/files".
Ideally we should write the args file in this directory however, for some reasons, we can't do it through the android_shell.py when for phone is not rooted. We are looking into it.

For  now, you can manually add storage permissions for the Mojo Shell app and the spinning cube should work.

**Comment by smklein** [comment]
Got it. Thanks! Maybe the README.md file should be updated with these instructions so no one else bumps into the same issue?

**Comment by nellyv** [comment]
Good idea! Done

**Comment by qsr** [comment]
I have a CL in progress to solve the issue correctly for M.
